code/game/objects/items.dm

```diff
@@ -353,7 +353,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 // afterattack() and attack() prototypes moved to _onclick/item_attack.dm for consistency
 
-/obj/item/proc/hit_reaction(mob/living/carbon/human/owner, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+/obj/item/proc/hit_reaction(mob/living/carbon/human/owner, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK, atom/movable/AM)

```

Comment:
I prefer @kevinz000 attempt.
put the new arg in second place, and update all calls.

If hit reaction had been perfect from the start, that would be the order.
Also, don't call it ```AM``` give it a descriptive name (like, ```reaction_source```)